src/Microsoft.AspNetCore.Sockets.Client/SendMessage.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public struct SendMessage
+    {
+        public MessageType Type { get; }
+        public byte[] Payload { get; }
+        public TaskCompletionSource<object> SendResult { get; }
```

Comment:
Just some questions. So this is just a Message but with the EOM bool replaced with a SendResult TCS? Is the TCS just used to report transport errors to the Connection.SendAsync() caller? Do Channels not have a built in mechanism for this like Pipelines?